### PR TITLE
Fix issue with nullable type mismatches

### PR DESCRIPTION
**Pull request by khalidabuhakmeh**
Nullable types are a beast. This generates a valid LINQ expression to handle them and to handle the conversion when there is a type mismatch. This only needs to happen when there is a type mismatch.

### CODE REVIEW

File: tests/Filter.Tests/Generic/IEnumerable`1ExtensionsTests.cs [hunk 1]
```diff
@@ -331,42 +396,6 @@ public void Should_filter_when_property_types_match_as_range_ushort()
                 }
             }
 
-            [Fact]
```
Comment (williambogaiv) [inline tests/Filter.Tests/Generic/IEnumerable`1ExtensionsTests.cs]: These tests should remain in their original location.

Comment (khalidabuhakmeh) [inline tests/Filter.Tests/Generic/IEnumerable`1ExtensionsTests.cs]: The issue with them being there, is that our test runner was counting them as duplicate tests on each run. So each test would be run for each subclass. 

> 3 classes * 3 tests = 9 tests

Is that what we actually want?

**Comment by williambogaiv** [comment]
:+1:

**Comment by williambogaiv** [comment]
image

**Pull request merged**